src/services/AsyncStorage.ts

```diff
@@ -24,13 +24,23 @@ interface sessionData {
 
 export const getCustomDisciplines = async (): Promise<string[]> => {
   const disciplines = await AsyncStorage.getItem('customDisciplines')
-  return disciplines ? JSON.parse(disciplines) : null
+  return disciplines ? JSON.parse(disciplines) : []
 }
 
 export const setCustomDisciplines = async (customDisciplines: string[]): Promise<void> => {
   await AsyncStorage.setItem('customDisciplines', JSON.stringify(customDisciplines))
 }
 
+export const deleteCustomDiscipline = async (customDiscipline: string): Promise<void> => {
+  const disciplines = await getCustomDisciplines()
+  const index = disciplines.indexOf(customDiscipline)
+  if (index === -1) {
+    return await Promise.reject(new Error('customDiscipline not available'))
```

Comment:
i think you can just throw the error instead of awaiting the reject. this avoids the creating of a new microtask
```suggestion
    throw new Error('customDiscipline not available')
```

Reply:
done